### PR TITLE
Refactor code search docs

### PR DESCRIPTION
**Pull request by jtibshirani**
This PR finishes the search docs reorganization started in #471.

Specific changes:
* Remove Search Types section (since this is not really a concept we have) and move its children pages to existing sections
* Improve the intro to "Search Query Syntax", other light edits
* Rename "Fuzzy Search" -> "Fuzzy Finder" for clarity
* Rename "Search Filters" -> "Search Filters Panel" and move it into "Search Query Syntax"
* Remove the "Exhaustive search" page, since we now recommend using Search Jobs.
* ~~Remove the search aggregations docs~~
* Simplify Features page, move config details to Search Configuration page

Closes SPLF-124

### CODE REVIEW

File: docs/code-search/features.mdx [hunk 1]
```diff
@@ -2,11 +2,15 @@
 
 <p className="subtitle">Learn and understand more about Sourcegraph's Code Search features and core functionality.</p>
 
+<LinkCards>
```
Comment (jtibshirani) [inline docs/code-search/features.mdx:5]: Originally, this was at the top of the "Search Query Syntax" page. That page is an important technical reference, and is already a bit intimidating to read, so I felt it was best not to interrupt flow with a marketing callout.

File: docs/code-search/queries/index.mdx [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
+A typical search consists of two parts:
+* A [search pattern](#-search-patterns) containing the terms you want to search, for example `println`
+* [Search filters](#filters-all-searches) that scope the search to certain repositories, languages, etc., for example `lang:java`
 
-<LinkCards>
-  <LinkCard href="https://about.sourcegraph.com/get-started?t=enterprise" imgSrc="https://sourcegraph.com/.assets/img/sourcegraph-mark.svg" imgAlt="Sourcegraph" title="Get single-tenant instance managed by Sourcegraph." description="Sign up and get a 30 day free trial for your team." />
-</LinkCards>
+For an exhaustive list of available search syntax, see the [search language reference](/code-search/queries/language).
```
[the remark this answers is not in the document]
Comment (jtibshirani) [reply docs/code-search/queries/index.mdx:9]: For now, I'll just rework this to say "for a graphical view of the search syntax, see..." Then we can follow-up with a PR unifying these and removing the railroad diagrams.

**Comment by jtibshirani** [comment]
I like your radical ideas! Some responses...

> Do we need a doc page for "Search Subexpressions"?

I see it being helpful (we do hear from customers crafting complex parenthesized expressions), and it doesn't hurt because it's tucked away under "Advanced Features". No strong opinion though, what do you think?

> Should we move "Symbol Search" from "Search Query Syntax" to "Advanced Features"?

I consider symbol search to be a core part of our offering, especially since we expose it in the search sidebar. I would be open to moving it to "Code Navigation" though, especially since we talk a lot about the symbols sidebar. What do you think?

> Remove the railroad diagrams?

I agree, I think we should fold the "Search Query Syntax" page and "Language Reference" pages together, simplify and remove the railroad diagrams. Let's tackle this in a follow-up PR as this one is already substantial. 

> Features: I think "Max Files Size" should not be listed as feature, but rather move to a dedicated config page?
> The sections for "Exclude files and directories" and "Multi-branch indexing" could be split it two, with the config part moving to a dedicated config page

I made a new section "Configuration Reference" and also linked to it from the admin docs. I moved "Max File Size" and "Shard merging" there, since they are quite low-level and don't fit well in "Features". I also moved the config details from "Exclude files" and "Multi-branch indexing" to that new page.

> "Suggestions" seems like table stakes to me. I am not sure if this should be listed as feature anymore.

+1 I will remove this section.

**Review by stefanhengl — APPROVED** [review]
Thanks for the changes!

File: docs/code-search/queries/index.mdx [hunk 3]
```diff
@@ -1,16 +1,16 @@
 # Search Query Syntax
 
-<p className="subtitle">This page describes search pattern syntax and filters available for Code Search.</p>
+<p className="subtitle">This page describes the query syntax for Code Search.</p>
 
-See the complementary [language reference](/code-search/queries/language) for a visual breakdown. A typical search pattern describes content or filenames to find across all repositories. At the most basic level, a search pattern can simply be a word like `hello`. See our [search patterns](#search-pattern-syntax) documentation for detailed usage. Queries can also include filters. For example, a typical search query will include a `repo:` filter that filters search results for a specific repository. See our [filters](#filters-all-searches) documentation for more examples.
+A typical search consists of two parts:
+* A [search pattern](#-search-patterns) containing the terms you want to search, for example `println`
```
Comment (MaedahBatool) [inline docs/code-search/queries/index.mdx:6]:
```suggestion
* A [search pattern](#search-patterns) containing the terms you want to search, for example `println`
```

**Review by MaedahBatool — APPROVED** [review]

**Comment by MaedahBatool** [comment]
Folks are we good to go to merge this PR to the working branch?

**Pull request merged**